java/client/src/main/java/glide/api/models/GlideString.java

```diff
@@ -0,0 +1,100 @@
+/** Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.api.models;
+
+import java.util.Arrays;
+import lombok.Getter;
+
+// TODO docs for the god of docs
+public class GlideString {
+
+    @Getter private byte[] bytes;
+    private String string = null;
+
+    /** Flag whether possibility to convert to string was checked. */
+    private boolean conversionChecked = false;
+
+    private GlideString() {}
+
+    public static GlideString of(String string) {
+        var res = new GlideString();
+        res.string = string;
+        res.bytes = string.getBytes();
```

Comment:
Specify the target charset since this can vary from system to system.
Probably should be UTF-8.